python/data_extraction/export_data.py

```diff
@@ -1,63 +1,78 @@
 from data_extraction import content_export
 from data_extraction import taxonomy_query, plek
-import json
-import sys
+from data_extraction import json_arrays
+from data_extraction.helpers import dig
 import functools
 from multiprocessing import Pool
+import gzip
+import json
+import os
+
 
+config_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), '..', 'config', 'data_export_fields.json')
 
-with open('config/data_export_fields.json') as json_data_file:
+with open(config_path) as json_data_file:
     configuration = json.load(json_data_file)
 
 
-content_links_generator = content_export.content_links_generator(
-    blacklist_document_types=configuration['blacklist_document_types']
-)
+def __transform_content(input_filename="data/content.json.gz",
+                        output_filename="data/filtered_content.json.gz",
+                        transform_function=lambda x: x):
```

Comment:
I'm not quite sure about the custom JSON parsing stuff, but I like what you've done here with transforming the file to generate a new file.

Reply:
json is hard to parse incrementally - i.e. without reading in the whole file in one go and then keeping the entire content-store in memory

Reply:
There look to be some libraries that talk about parsing JSON in a stream, maybe we can use one of them. I've used streaming XML parsers in the past, so hopefully the same thing exists for JSON. It would be good to have a solution that is independent of the formatting of the file. This is ok as a starting point though.